sdk/include/ndk/pstypes.h

```diff
@@ -972,6 +972,23 @@ typedef struct _JOB_SET_ARRAY
     ULONG Flags;
 } JOB_SET_ARRAY, *PJOB_SET_ARRAY;
 
+//
+// Process Quota Type
+//
+typedef enum _PS_QUOTA_TYPE
+{
+    PsNonPagedPool = 0,
+    PsPagedPool,
+    PsPageFile,
+#if (NTDDI_VERSION >= NTDDI_LONGHORN)
+    PsWorkingSet,
+#endif
+#if (NTDDI_VERSION == NTDDI_LONGHORN)
```

Comment:
What's the logic here? `PsCpuRate` was removed after Vista?

Reply:
According to Geoff Chappell https://www.geoffchappell.com/studies/windows/km/ntoskrnl/api/ps/psquota/type.htm :
```
Value | Name | Versions
  --  |  --  |   --
  0  | PsNonPagedPool | 5.1 and higher
  1  | PsPagedPool    | 5.1 and higher
  2  | PsPageFile     | 5.1 and higher
  3  | working set pages | 6.0 and higher
**4  | CPU rate limit    | 6.0 to 6.2 ** <--- This one
  3 (5.1 to 5.2);
  5 (6.0 to 6.2);  | PsQuotaTypes | 5.1 and higher
  4
```